src/main/java/org/jenkinsci/plugins/workflow/cps/SnippetizerLink.java

```diff
@@ -75,32 +76,12 @@ public final String getDisplayUrl() {
 
         StaplerRequest req = Stapler.getCurrentRequest();
         if (req == null) {
-            return u;
+            throw new IllegalStateException("Can't get display URL without Stapler context.");
         }
 
         Item i = req.findAncestorObject(Item.class);
 
-        StringBuilder toAppend = new StringBuilder();
-
-        toAppend.append(req.getContextPath());
-
-        if (!req.getContextPath().endsWith("/")) {
-            toAppend.append("/");
-        }
-
-        if (i == null) {
-            toAppend.append(u);
-        } else {
-            toAppend.append(i.getUrl());
-
-            if (!i.getUrl().endsWith("/")) {
-                toAppend.append("/");
-            }
-
-            toAppend.append(u);
-        }
-
-        return toAppend.toString();
+        return Paths.get( req.getContextPath(), i == null ? "" : i.getUrl(), u).toString();
```

Comment:
Using `Paths` to concatenate class to handle slashes - will remove or add as needed.

Reply:
I am little wary of using `Path` for URLs, would it not result in file system-specific behavior? Maybe fine, I just haven't seen it used in this kind of context.

Reply:
And you would be correct. 😄